File: importa_clusters.js
var fs = require('fs');
var recursive = require('recursive-readdir');
var db = require('./db');
var constants = require('./constants');

/*
console.log('Excluindo commits_issues');
db.query('delete from clusters_entidades', function (err, results) {
	if (err) { throw err; }
	db.query('delete from clusters', function (err, results) {
		if (err) { throw err; }
*/
		recursive(__dirname, function (err, files) {
			import_files(files);
		});
/*
	});
});
*/

function import_files(files) {
	var arr, repository, file, name, content, regex, i;
	regex = /subgraph (.+) {/;
	for (i = 0; i < files.length; i += 1) {
		if (/\.dot$/.test(files[i])) {
			arr = files[i].replace(__dirname, '').split('/');
			repository = constants.repositorioMap[arr[arr.length-2]];
			file = arr[arr.length-1];
			content = fs.readFileSync(files[i],{ encoding: 'utf8'}).split('\n');
			for (var j = 0; j < content.length; j++) {
				if (regex.test(content[j])) {
					name = regex.exec(content[j])[1];
					db.query('insert into clusters(repositorio, arquivo, nome) values (?,?,?)',
						[repository, file, name],
						function (err, results) {
							if (err) throw err;
						}
					);
				}
			};
		}
	}
}

File: importa_clusters.py
#!/usr/bin/python

import os
import re
import argparse
from db import Db
import constants
from glob import glob

parser = argparse.ArgumentParser()
parser.add_argument("-r", "--repository", default="siop")
args = parser.parse_args()

files = [y for x in os.walk('.') for y in glob(os.path.join(x[0], '*.dot'))]

previous_graphs = {}

db = Db()

repository = constants.repository_map[args.repository]

print "Excluindo clusters_entidades"
db.execute("delete from clusters_entidades where entidade in (select id from entidades where repositorio = %s)", (repository,))
print "Excluindo clusters"
db.execute("delete from clusters where grafo in (select id from grafos where repositorio = %s)", (repository,))
print "Excluindo grafos"
db.execute("delete from grafos where repositorio = %s", (repository,))

print "Incluindo grafos"
for file in files:
	if re.search('\.dot$', file):
		file_name = file.rpartition('/')[2]
		file_repository = None
		for k in reversed(sorted(constants.repository_map.keys())):
			if file_name.startswith(k):
				file_repository = constants.repository_map[k]
				break
		if file_repository != repository:
			continue
		graph_id = db.insert('insert into grafos(repositorio, arquivo) values (%s,%s)', 
			(file_repository,file_name))
	with open(file) as f:
		print file
		for line in f:
			match = re.search('subgraph (.+) {', line)   			
			if match:
				cluster_name = match.group(1)
				cluster_id = db.insert('insert into clusters(grafo, nome) values (%s,%s)', 
					(graph_id, cluster_name))
			match = re.search('\"(\d+)\"\[', line)
			if match:
				db.insert('insert into clusters_entidades(cluster, entidade) values (%s,%s)', 
					(cluster_id, match.group(1)))

db.commit()
db.close()
